File: WorldFeed/Server/src/Services/Upload/WorldFeed.Upload.API/Client/Clients/ExecuteClient.cs
namespace WorldFeed.Upload.Client.Clients
{
    using System;
    using System.Threading.Tasks;
    using WorldFeed.Upload.API.Web;
    using WorldFeed.Upload.Application.Query;
    using WorldFeed.Upload.Application.Requesters;
    using WorldFeed.Upload.Client.Requesters;

    public class ExecuteClient : IExecuteClient
    {
        private readonly IExecuteRequester executeRequester;

        public ExecuteClient(IExecuteRequester executeRequester)
        {
            this.executeRequester = executeRequester;
        }

        public Task<ITwitterResult<T>> AdvanceRequestAsync<T>(Action<ITwitterRequest> configureRequest)
            where T : class
        {
            return this.executeRequester.RequestAsync<T>(configureRequest);
        }

        public Task<ITwitterResult> AdvanceRequestAsync(Action<ITwitterRequest> configureRequest)
        {
            return this.executeRequester.RequestAsync(configureRequest);
        }

        public Task<ITwitterResult<T>> RequestAsync<T>(Action<ITwitterQuery> configureQuery)
            where T : class
        {
            return this.executeRequester.RequestAsync<T>(configureQuery);
        }

        public Task<ITwitterResult> RequestAsync(Action<ITwitterQuery> configureQuery)
        {
            return this.executeRequester.RequestAsync(configureQuery);
        }

        public Task<ITwitterRequest> PrepareTwitterRequestAsync(Action<ITwitterQuery> configureQuery)
        {
            return this.executeRequester.PrepareTwitterRequestAsync(configureQuery);
        }

        public Task<ITwitterRequest> PrepareTwitterRequestAsync(Action<ITwitterRequest> configureRequest)
        {
            return this.executeRequester.PrepareTwitterRequestAsync(configureRequest);
        }
    }
}

File: WorldFeed/Client/src/app/worldfeed/Client/Clients/ExecuteClient.ts
import {ITwitterResult} from "../../../core/Core/Web/TwitterResult";
import {ITwitterRequest} from "../../../core/Public/Models/Interfaces/ITwitterRequest";
import {ITwitterQuery} from 'src/app/core/Public/Models/Interfaces/ITwitterQuery';
import {IExecuteClient} from "../../../core/Public/Client/Clients/IExecuteClient";
import {IExecuteRequester} from "../../../core/Public/Client/Requesters/IExecuteRequester";

export class ExecuteClient implements IExecuteClient {
  private readonly _executeRequester: IExecuteRequester;

  constructor(executeRequester: IExecuteRequester) {
    this._executeRequester = executeRequester;
  }

  public advanceRequestAsync<T>(configureRequest: (twitterRequest: ITwitterRequest) => void): Promise<ITwitterResult<T>> {
    return this._executeRequester.requestAsync<T>(configureRequest);
  }

  public advanceRequestAsync(configureRequest: (twitterRequest: ITwitterRequest) => void): Promise<ITwitterResult> {
    return this._executeRequester.requestAsync(configureRequest);
  }

  public requestAsync<T>(configureQuery: (twitterQuery: ITwitterQuery) => void): Promise<ITwitterResult<T>> {
    return this._executeRequester.requestAsync<T>(configureQuery);
  }

  public requestAsync(configureQuery: (twitterQuery: ITwitterQuery) => void): Promise<ITwitterResult> {
    return this._executeRequester.requestAsync(configureQuery);
  }

  public PrepareTwitterRequestAsync(configureQuery: (twitterQuery: ITwitterQuery) => void): Promise<ITwitterRequest> {
    return this._executeRequester.prepareTwitterRequestAsync(configureQuery);
  }

  public PrepareTwitterRequestAsync(configureRequest: (twitterRequest: ITwitterRequest) => void): Promise<ITwitterRequest> {
    return this._executeRequester.prepareTwitterRequestAsync(configureRequest);
  }
}
